packages/blockchain-link/src/workers/solana/fee.ts

```diff
@@ -35,31 +54,40 @@ export const getBaseFee = async (api: Connection, message: Message) => {
 
 // More about Solana priority fees here:
 // https://solana.com/developers/guides/advanced/how-to-use-priority-fees#how-do-i-estimate-priority-fees
-export const getPriorityFee = async (api: Connection, message: Message) => {
-    const affectedAccounts = message
-        .getAccountKeys()
-        .staticAccountKeys.filter((_, i) => message.isAccountWritable(i))
-        .map(key => key.toString());
+export const getPriorityFee = async (
+    api: Rpc<GetRecentPrioritizationFeesApi>,
+    compiledMessage: CompiledTransactionMessage,
+) => {
+    const message = decompileTransactionMessage(compiledMessage);
+    const affectedAccounts = new Set<Address>(
+        message.instructions
+            .flatMap(instruction => instruction.accounts ?? [])
+            .filter(({ role }) => isWritableRole(role))
+            .map(({ address }) => address),
+    );
 
-    const recentFees = await api.getRecentPrioritizationFees({
-        lockedWritableAccounts: affectedAccounts.map(a => new PublicKey(a)),
-    });
+    const recentFees = await api.getRecentPrioritizationFees(Array.from(affectedAccounts)).send();
 
     const computeUnitLimit = DEFAULT_COMPUTE_UNIT_LIMIT;
 
-    const networkPriorityFee = recentFees.map(a => a.prioritizationFee).sort((a, b) => b - a)[
-        Math.floor(recentFees.length / 4)
-    ]; // 25th percentile because many 0 priority fees are expected
+    const networkPriorityFee = recentFees
+        .map(a => a.prioritizationFee)
+        .sort((a, b) => Number(b - a))[Math.floor(recentFees.length / 4)]; // 25th percentile because many 0 priority fees are expected
 
-    const computeUnitPrice = new BigNumber(
-        Math.max(networkPriorityFee, DEFAULT_COMPUTE_UNIT_PRICE_MICROLAMPORTS),
-    );
+    const computeUnitPrice =
+        networkPriorityFee > DEFAULT_COMPUTE_UNIT_PRICE_MICROLAMPORTS
+            ? networkPriorityFee
+            : DEFAULT_COMPUTE_UNIT_PRICE_MICROLAMPORTS;
 
-    const fee = computeUnitPrice
+    const fee = new BigNumber(computeUnitPrice.toString())
         .times(10 ** -6) // microLamports -> Lamports
-        .times(computeUnitLimit)
+        .times(computeUnitLimit.toString())
         .decimalPlaces(0, BigNumber.ROUND_UP)
-        .toNumber();
+        .toString();
```

Comment:
```suggestion
        .toString(10);
```
This is to prevent BigNumber.toString outputting large numbers in scientific notation